### PR TITLE
chore(profiling): Use Pagefilter[datetime] on profiling chart

### PR DESCRIPTION
**Pull request by Zylphrex**

### CODE REVIEW

**Review by JonasBa — APPROVED** [review]
Love it @Zylphrex 🚀

File: static/app/views/profiling/content.tsx [hunk 1]
```diff
@@ -93,7 +93,12 @@ function ProfilingContent({location, selection}: ProfilingContentProps) {
                   </Alert>
                 )}
                 <ProfilingScatterChart
-                  {...dateSelection}
+                  datetime={{
```
Comment (JonasBa) [inline static/app/views/profiling/content.tsx:96]: Small nit: this object is creating a new reference each time. Do we need to cast to null here, or can we just pass datetime={dateSelection} and handle it in the ProfilingScatterChart? This way we have the consistent type signature + no new reference.

File: static/app/views/profiling/content.tsx [hunk 2]
```diff
@@ -93,7 +93,12 @@ function ProfilingContent({location, selection}: ProfilingContentProps) {
...
+                    period: dateSelection.statsPeriod ?? null,
+                    start: dateSelection.start ?? null,
+                    end: dateSelection.end ?? null,
```
Comment (JonasBa) [inline static/app/views/profiling/content.tsx:99]: `s/dateSelection/datetime`

File: static/app/views/profiling/content.tsx (hunk 1 above)
Comment (Zylphrex) [reply static/app/views/profiling/content.tsx]: I opted to use the `datetime` directly from the page filters so that the types are aligned

**Pull request merged**